catalyst_voices/uikit_example/lib/widgets/session_header.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:catalyst_voices/widgets/buttons/voices_filled_button.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_blocs/catalyst_voices_blocs.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+/// Displays current session state and a button to toggle it to a next state.
+class SessionHeader extends StatelessWidget {
+  const SessionHeader();
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<SessionBloc, SessionState>(
+      builder: (context, state) {
+        return switch (state) {
+          VisitorSessionState() => const _VisitorSessionHeader(),
+          GuestSessionState() => const _GuestSessionHeader(),
+          ActiveUserSessionState() => const _ActiveUserSessionHeader(),
+        };
+      },
+    );
+  }
+}
+
+class _VisitorSessionHeader extends StatelessWidget {
+  const _VisitorSessionHeader();
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        VoicesFilledButton(
+          child: Text('Get Started'),
+          onTap: () =>
+              context.read<SessionBloc>().add(const NextStateSessionEvent()),
+        ),
+        const SizedBox(width: 12),
+        VoicesFilledButton(
+          child: Text('Visitor'),
+        ),
+      ],
+    );
+  }
+}
+
+class _GuestSessionHeader extends StatelessWidget {
+  const _GuestSessionHeader();
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        VoicesFilledButton(
+          trailing: Icon(CatalystVoicesIcons.lock_open),
+          child: Text('Unlock'),
+          onTap: () =>
+              context.read<SessionBloc>().add(const NextStateSessionEvent()),
+        ),
+        const SizedBox(width: 12),
+        VoicesFilledButton(
+          child: Text('Guest'),
+        ),
+      ],
+    );
+  }
+}
+
+class _ActiveUserSessionHeader extends StatelessWidget {
+  const _ActiveUserSessionHeader();
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        VoicesFilledButton(
+          trailing: Icon(CatalystVoicesIcons.lock_closed),
+          child: Text('Lock'),
+          onTap: () =>
+              context.read<SessionBloc>().add(const NextStateSessionEvent()),
```

Comment:
probably not for this PR but it should be cool to have extension on `Bloc` like `read` or smth else so you could just do smth like this

```dart
SessionBloc.read(context).add(const NextStateSessionEvent());
```
or
```dart
SessionBloc.of(context).add(const NextStateSessionEvent());
```

Reply:
The method would need to be static, statics cannot be inherited nor provided by extension. Then for every bloc it would need to be done separately. Imho too much complication for what it's worth.

We have BlocProvider.of(context) that returns you a bloc, the problem is it's generic so you need to call it like `BlocProvider.of<YourBloc>(context)`.

Reply:
I see :( 
but yeah, agree, not worth it to put static methods everywhere

Reply:
For the user management, I'm not sure what exactly will be needed there. I can assume that there will be two states: User / No user. A user would have it's profile (name, etc.).

If you agree with this assumption I'll add it in next PR